src/context/movie-details-context/MovieDetailsContextProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { createContext, FC, useContext, useState } from 'react'
+import { IMovieDetailsContext } from './movieDetailsContextTypes'
+import { IMovieItem } from '../../pages/Home/components/movie-card'
+
+const MovieDetailsContext = createContext({
+    movieInfo: {} as IMovieItem,
+    getMovieDetails: (movie: IMovieItem) => console.log(movie),
+    movieDetailsState: false,
+    hideMovieDetails: () => console.log('false')
```

Comment:
It is good idea to throw Errors if someone will try use until context provider is mounted:

```tsx
hideMovieDetails: () => trhow Error("Do not use context out of context provider")
```

Reply:
done. changed for all contexts